src/stumpy_core/canvas.cr

```diff
@@ -34,5 +34,36 @@ module StumpyCore
         @height == other.height &&
         @pixels == other.pixels
     end
+    
+    def paste(canvas, x, y)
+      #StumpyPNG does not handle going out of bound well so we have to be very careful with the paste
+
+      #detect left side first
+      min_x = (x < 0 ? x.abs : 0)
```

Comment:
I think `min_x = x < 0 ? -x : 0` would be a little faster,
because `.abs` is implemented as `x < 0 ? -x : x`